### ISSUE TITLE
How to define a variable with different value for each tower under MirroredStrategy

### ISSUE THREAD

**Issue by CtfGo** [tensorflow/tensorflow#24875]
I tried to train model on multi-gpus with tf estimator and MirroredStrategy. And I found that variables of the model are wrapped as MirroredVariables which are always in sync among towers.

However, there is a variable that needs to have different value for each tower on every step of training, How to define such variable ? that is, a variable with locality T, refers to (https://www.tensorflow.org/api_docs/python/tf/contrib/distribute/DistributionStrategy)

**Comment by jvishnuvardhan** [comment]
@CtfGo Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support. Github is mainly for addressing bugs in installation and performance. Responses to your questions will help others also if you post them on Stackoverflow. Thanks!

**Comment by guptapriya** [comment]
@CtfGo @x10000year  can you describe the use case, why you need variables which are not synced across towers? Also would you want these variables to be synced at all, or never synced?

**Comment by x10000year** [comment]
@guptapriya 

One of my use cases is to implement truncated back propagation for a unrolled rnn. In this case, for every training step and every tower we read only n steps from a sequence, and compute n steps of the unrolled rnn. The computation starts with the last rnn states from the same sequence, and at the end stores the rnn states into local variables.  The local/temporary variables are never synced. For each tower we have a custom data input op that guarantees every data sequence is processed by the same tower.

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by isaprykin** [comment]
You could try to create variables with `VariableSynchronization.ON_READ`.  If that doesn't work, you could look into creating a new distribution strategy which implements your logic of syncing them (or in your case not syncing them at all?).  Please re-open if those methods don't help.
